azurerm/helpers/azure/spring_cloud.go

```diff
@@ -0,0 +1,47 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+)
+
+func ValidateSpringCloudName(i interface{}, k string) (_ []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		return nil, append(errors, fmt.Errorf("expected type of %s to be string", k))
+	}
+
+	// The name attribute rules are :
+	// 1. can contain only lowercase letters, numbers and hyphens.
+	// 2. The first character must be a letter.
+	// 3. The last character must be a letter or number
+	// 3. The value must be between 4 and 32 characters long
+
+	if len(v) < 4 || len(v) > 32 {
+		errors = append(errors, fmt.Errorf("%s must be between 4 and 32 characters long", k))
+	} else {
+		if m, _ := validate.RegExHelper(i, k, `^([a-z])([a-z\d-]{0,30})([a-z\d])$`); !m {
```

Comment:
This regex is also checking the length?

Reply:
@katbyte  This regex is to ensure:
1. starts with lower case character
2. ends with lower case character or number
3. can only contains lower case character and number and '-'